src/Form/Template.php

```diff
@@ -256,6 +256,15 @@ public function getDonationIntroductionContentPosition() {
 	 * @return Receipt
 	 */
 	public function getReceiptDetails( $donationId ) {
-		return new Receipt( $donationId );
+		$receipt = new Receipt( $donationId );
+
+		/**
+		 * fire the action for receipt object.
+		 *
+		 * @since 2.7.0
+		 */
+		do_action( 'give_new_receipt', $receipt );
```

Comment:
Why is the receipt being constructed in two different places? Here and in the `Sequoia` class.

Reply:
@JasonTheAdams This is an abstract class, so I added a default function. Any template have ability to overwrite it.